File: backend/aperio-api/src/infrastructure/adapters/ai/prompts/entity_prompt.py
ENTITY_SYSTEM_PROMPT = """You are an entity extractor for a plastic recycling supply chain system called Aperio.

Given a user message about a material transaction, extract the following entities into JSON:

Required fields:
- "material": One of "PET", "HDPE", "PP", "LDPE", "PVC", "mixed"
- "quantity_kg": Numeric quantity in kilograms

Optional fields:
- "vendor": Vendor/supplier name
- "date": Date of transaction (use relative terms like "today", "yesterday", or ISO format)
- "stage": Processing stage (sorting, washing, shredding, extrusion, pelletizing)
- "loss_kg": Material loss in kg
- "batch_id": Batch identifier if mentioned
- "notes": Any additional notes

Respond ONLY with valid JSON. Example:
{"material": "PET", "quantity_kg": 300, "vendor": "Vendor A", "date": "today"}
"""

File: frontend/src/infrastructure/ai/prompts/entityPrompt.ts
import type { AIClient } from '../provider';
import type { ParsedIntent } from '@/shared/types';

const ENTITY_SYSTEM_PROMPT = `You are an entity extractor for a plastic recycling platform.
Extract structured data from the user's message. Return a JSON object with these fields:
- intent: the action type
- material: PET, HDPE, PP, LDPE, PVC, or mixed
- quantity_kg: number
- vendor: string (optional)
- date: ISO 8601 date string
- stage: collection, sorting, processing, output, or dispatch (optional)
- loss_kg: number (optional)
- batch_id: string (optional)
- notes: string (optional)`;

export async function extractEntities(
  ai: AIClient,
  message: string,
  intentType: string,
): Promise<ParsedIntent> {
  const prompt = `Intent type: ${intentType}\nUser message: "${message}"`;
  const raw = await ai.complete(prompt, ENTITY_SYSTEM_PROMPT);
  try {
    return JSON.parse(raw) as ParsedIntent;
  } catch {
    throw new Error('Failed to parse entity extraction response');
  }
}
